### PR TITLE
adjust contribution guide

### PR DESCRIPTION
**Pull request by johnnyaug**
As we are receiving more and more contributions from the community, we want developers to be able to suggest their contribution idea before coding it, to avoid them wasting their time.

I went over other OSS projects, and saw that usually their contribution guide says something like "large PRs should have a GitHub (or Jira) ticket attached to them". Some of them also require contacting the team before opening pull requests.

I suggest the following change to the guidelines.

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -10,24 +10,21 @@ has_children: false
...
-* Submit any security issues directly to [security@treeverse.io](mailto:security@treeverse.io)
-
-
-## Getting Started
-
-Want to report a bug or request a feature? Please [open an issue](https://github.com/treeverse/lakeFS/issues/new)
-
-Working on your first Pull Request? You can learn how from this free series, [How to Contribute to an Open Source Project on GitHub](https://app.egghead.io/playlists/how-to-contribute-to-an-open-source-project-on-github).
+* Submit any security issues directly to [security@treeverse.io](mailto:security@treeverse.io).
+* Contributions should have an associated [Github issue](https://github.com/treeverse/lakeFS/issues/).
```
Comment (nopcoder) [inline docs/contributing.md:25]: s/Github/GitHub/g

File: docs/contributing.md [hunk 2]
```diff
@@ -10,24 +10,21 @@ has_children: false
 
 Thank you for your interest in contributing to our project. Whether it's a bug report, new feature, correction, or additional documentation, we greatly value feedback and contributions from our community.
 
-Please read through this document before submitting any issues or pull requests to ensure we have all the necessary information to effectively respond to your bug report or contribution..
+Please read through this document before submitting any issues or pull requests to ensure we have all the necessary information to effectively respond to your bug report or contribution.
 
-If you don't know where to start, please [join our community on Slack](https://lakefs.io/slack) and ask us. We will help you get started!
+Don't know where to start? [Join our community on Slack](https://lakefs.io/slack) and we will help you get started. We also recommend this [free series](https://app.egghead.io/playlists/how-to-contribute-to-an-open-source-project-on-github){:target="_blank"} about contributing to OSS projects.
```
Comment (arielshaqed) [inline docs/contributing.md:15]: Mention the dev and help channels?

**Review by arielshaqed — APPROVED** [review]
Great improvements all around!

File: docs/contributing.md [hunk 3]
```diff
@@ -10,24 +10,21 @@ has_children: false
...
 Before you get started, we ask that you:
 
 * Check out the [code of conduct](https://github.com/treeverse/lakeFS/blob/master/CODE_OF_CONDUCT.md).
 * Sign the [lakeFS CLA](https://cla-assistant.io/treeverse/lakeFS) when making your first pull request (individual / corporate)
-* Submit any security issues directly to [security@treeverse.io](mailto:security@treeverse.io)
-
-
-## Getting Started
-
-Want to report a bug or request a feature? Please [open an issue](https://github.com/treeverse/lakeFS/issues/new)
```
Comment (guy-har) [inline docs/contributing.md:28]: Think we should leave this or something similar

**Review by guy-har — APPROVED** [review]

File: docs/contributing.md (hunk 3 above)
Comment (johnnyaug) [reply docs/contributing.md:28]: It was replaced by the 4th bullet on the Getting started list.

**Pull request merged**